keps/draft-20180426-security-conformance.md

```diff
@@ -0,0 +1,532 @@
+  (`default` or explicitly specified in `ServiceAccountName`),
+  should have all capabilities dropped.
+
+###### Test Recommendation
+
+  This test requires a container image with the programs to verify Linux capabilities.
+
+  The test code consumes Kubernetes API as a serviceaccount/user with privileges:
+
+  - Create/Delete a namespace
+  - Create a ReplicaSet in the new namespace
+
+
+  Test setup
+
+  - Create a new namespace
+
+
+  Test steps
+
+  - Create a ReplicaSet, in PodTemplate/SecurityContext, add capabilities for test in `capabilities`
+  - Poll the ReplicaSet and the namespace, and a new Pod should be created
+  - Check the output of the Pod (can be log) and make sure no capabilities is added
+
+
+  Test teardown
+
+  - Delete the namespace
+
+#### Limited Volume Types
+
+  *Conformance Profile: Restricted*
+
+  Pod creation with volume types outside of
+  `configMap`, `secret`, `downwardAPI`, `emptyDir`, `persistentVolumeClaim`, `projected`
+  should fail if there's no explicit RBAC associated with the Pod service account
+  (`default` or explicitly specified in `ServiceAccountName`).
+
+###### Test Recommendation
+
+  The test code consumes Kubernetes API as a serviceaccount/user with privileges:
+
+  - Create/Delete a namespace
+  - Create a ReplicaSet in the new namespace
+
+
+  Test setup
+
+  - Create a new namespace
+
+
+  Test steps
+
+  - Create a ReplicaSet, with volumes of types of
+    `configMap`, `secret`, `downwardAPI`, `emptyDir`, `persistentVolumeClaim`, `projected`
+  - Poll the ReplicaSet and the namespace, and a new Pod should be created
+  - Create a ReplicaSet, with a volume with type none of
+    `configMap`, `secret`, `downwardAPI`, `emptyDir`, `persistentVolumeClaim`, `projected`
+  - Poll the ReplicaSet, should got `ReplicaFailure` condition from `status`, and no Pod is created
+
+
+  Test teardown
+
+  - Delete the namespace
+
+#### Run as root
+
+  *Conformance Profile: Isolated*
+
+  Containers in Pod created without explicit RBAC associated with the Pod service account,
+  should run as non-root user and group.
+
+  _Note: this may change if a better isolation runtime (e.g. VM) is used for the container._
+
+###### Test Recommendation
+
+  This test requires a container image with the programs (e.g. `id`) to get current UID and GID.
+
+  The test code consumes Kubernetes API as a serviceaccount/user with privileges:
+
+  - Create/Delete a namespace
+  - Create a ReplicaSet in the new namespace
+
+
+  Test setup
+
+  - Create a new namespace
+
+
+  Test steps
+
+  - Create a ReplicaSet
+  - Poll the ReplicaSet and the namespace, and a new Pod should be created
+  - Check the output of the Pod (can be log) and make sure UID and GID are not zero
+
+
+  Test teardown
+
+  - Delete the namespace
+
+#### Network Isolation
+
+  *Conformance Profile: Isolated*
+
+  Ingress/Egress are blocked for all traffic for all Pods by default in namespaces,
+  except the Egress to access `kube-dns`.
```

Comment:
I'm not sure this makes sense as the default option. But maybe someone from networking has more opinions.

Reply:
I'm not sure either. But I feel reasonable, as all workloads are running on the same overlay network - they are logically connected together. There are bunch of attacking techniques over network.